ext/pdo_firebird/firebird_driver.c

```diff
@@ -1201,6 +1278,18 @@ static int pdo_firebird_handle_factory(pdo_dbh_t *dbh, zval *driver_options) /*
 		dbh->password = pestrdup(vars[5].optval, dbh->is_persistent);
 	}
 
+	H->in_manually_txn = 0;
+	H->is_writable_txn = pdo_attr_lval(driver_options, PDO_FB_WRITABLE_TRANSACTION, 1);
+	zend_long txn_isolation_level = pdo_attr_lval(driver_options, PDO_FB_TRANSACTION_ISOLATION_LEVEL, PDO_FB_REPEATABLE_READ);
+	if (txn_isolation_level == PDO_FB_READ_COMMITTED ||
+		txn_isolation_level == PDO_FB_REPEATABLE_READ ||
+		txn_isolation_level == PDO_FB_SERIALIZABLE
+	) {
+		H->txn_isolation_level = txn_isolation_level;
+	} else {
+		H->txn_isolation_level = PDO_FB_REPEATABLE_READ;
```

Comment:
Should I make it an error?

Reply:
I took a closer look, and the ret value is a mess.

I would like to leave this as is for now and fix it all together in a separate pull request.

Reply:
This value is set by users, so any value that is an int will come here. Therefore, it is quite possible that values ​​other than the three constant values ​​will be passed.

factory may need validation and errors as same as set attr.

Reply:
I would possibly throw a ValueError as trying to use an invalid transaction isolation mode is a programming error.

Reply:
Both the constructor and set attr have been fixed to cause a value error.
https://github.com/php/php-src/pull/12815/commits/b7d932f0d9325ed4c306071970949139fb4938e9

However, I am a little confused about whether it would be better to write `zend_restore_error_handling` in pdo core.

Reply:
Ah, `zend_value_error` is better. I'll fix it.

Reply:
done